src/lib/y2packager/system_packages.rb

```diff
@@ -51,9 +51,7 @@ def select
     #
     def repo_ids(urls)
       repo_ids = Yast::Pkg.SourceGetCurrent(true)
-      repo_ids.each_with_object([]) do |i, list|
-        list << i if urls.include?(Yast::Pkg.SourceGeneralData(i)["url"])
-      end
+      repo_ids.select { |i| urls.include?(Yast::Pkg.SourceGeneralData(i)["url"]) }
```

Comment:
when I see this just small note. In general I do not like much when method and local variable has identical name as in some cases it can be quite confusing (not mentioning clobbering). so what e.g. renaming local variable to `all_ids` ?

Reply:
oh, good point, I overlooked that...